packages/react-draggable-dialog/src/components/DraggableDialog/utils.ts

```diff
@@ -0,0 +1,77 @@
+import { Modifier } from '@dnd-kit/core';
+import { restrictToWindowEdges } from '@dnd-kit/modifiers';
+import {
+  DraggableDialogProps,
+  DraggableMargin,
+  DraggableMarginViewport,
+} from './DraggableDialog.types';
+
+type RestrictToMarginModifierOptions = {
+  margin: Required<DraggableMarginViewport>;
+} & Pick<DraggableDialogProps, 'keepInViewport'>;
+
+type RestrictToMarginModifier = (
+  options: RestrictToMarginModifierOptions
+) => Modifier;
+
+export const restrictToMarginModifier: RestrictToMarginModifier =
+  ({ margin, keepInViewport }) =>
+  ({ windowRect, transform, ...modifier }) => {
+    if (!keepInViewport || !windowRect) {
+      return transform;
+    }
+
+    return restrictToWindowEdges({
+      ...modifier,
+      transform,
+      windowRect: {
+        width: windowRect.width - margin.start - margin.end,
+        height: windowRect.height - margin.top - margin.bottom,
+        top: windowRect.top + margin.top,
+        right: windowRect.right + margin.end,
+        bottom: windowRect.bottom + margin.bottom,
+        left: windowRect.left + margin.start,
+      },
+    });
+  };
+
+export const getParsedDraggableMargin = (
+  margin: DraggableMargin | undefined
+): Required<DraggableMarginViewport> => {
+  const defaultMargin = {
```

Comment:
this doesn't need to be re-created on every invocation. can we move defaults constant to the module scope ?